test_app/integration_tests/android_mocks/src/main/java/com/voxeet/asserts/AudioPreviewAsserts.kt

```diff
@@ -41,146 +49,186 @@ class AudioPreviewAsserts : MethodDelegate {
-            )
-        } else {
-            throw KeyNotFoundException("hasRun key not found")
-        }
+        )
     }
 
     private fun assertReleaseArgs(args: Map<String, Any>?) {
-        val mockHasRun = VoxeetSDK.audio().local.preview().releaseHasRun
-        if (args!!.containsKey("hasRun")) {
-            AssertUtils.compareWithExpectedValue(
-                mockHasRun,
-                args!!["hasRun"],
+        val preview = VoxeetSDK.audio().local.preview()
+        AssertUtils.compareWithExpectedValue(
+                preview.releaseRunCount > 0,
+                true,
                 "audio preview release not run"
-            )
-        } else {
-            throw KeyNotFoundException("hasRun key not found")
-        }
+        )
     }
 
     private fun getStatus() {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.statusHasRun = true;
+        preview.statusRunCount = 0;
+        preview.statusReturn = recorderStatuses.toMutableList()
     }
 
     private fun getCaptureMode() {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.getCaptureModeHasRun = true;
+        preview.captureModeReturn.add(AudioCaptureMode.unprocessed());
+        for (voiceFont in voiceFonts) {
+            for (noiseReduction in noiseReductions) {
+                preview.captureModeReturn.add(AudioCaptureMode.standard(noiseReduction, voiceFont));
+            }
+        }
     }
 
     private fun setCaptureMode(args: Map<String, Any>?) {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.setCaptureModeHasRun = true
-        preview.setCaptureModeArgs = args
+        preview.captureModeArgs.clear()
     }
 
     private fun record(args: Map<String, Any>?) {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.recordHasRun = true
-        preview.recordArgs = args
+        preview.recordArgs.clear()
+        preview.recordReturn = mutableListOf(true, true)
     }
 
     private fun play(args: Map<String, Any>?) {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.playHasRun = true
-        preview.playArgs = args
+        preview.playModeArgs.clear()
+        preview.playModeReturn = mutableListOf(true, true)
     }
 
     private fun cancel() {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.cancelHasRun = true
+        preview.cancelRunCount = 0
+        preview.cancelReturn = mutableListOf(true)
     }
 
     private fun release() {
         val preview = VoxeetSDK.audio().local.preview()
-        preview.releaseHasRun = true
+        preview.releaseRunCount = 0
+        preview.releaseReturn = mutableListOf(true)
+    }
+
+    private fun emitStatusChangedEvents() {
+        val executorService = Executors.newSingleThreadExecutor()
+        val statuses = recorderStatuses.toMutableList()
+        var sendStatusWithDelay = { -> }
+        sendStatusWithDelay = { ->
```

Comment:
Line 179 declares the variable pointing to a closure. Line 180 assigns the closure to it. I have done this because `sendStatusWithDelay` is a recursive closure (ie. it calls it self). If I declare and assign the recursive closure in one line, Android studio is complaining that the closure does not exist on the recursive call.